jenkins/stage-maven-release/JenkinsFile

```diff
@@ -0,0 +1,112 @@
+pipeline {
+    agent {
+            docker {
+                label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                image 'opensearchstaging/ci-runner:al2-x64-arm64-jdk14-node10.24.1-cypress6.9.1-20211005'
+                // Unlike freestyle docker, pipeline docker does not login to the container and run commands
+                // It use executes which does not source the docker container internal ENV VAR
+                args '-e JAVA_HOME=/usr/lib/jvm/adoptopenjdk-14-hotspot'
+                alwaysPull true
+            }
+    }
+    environment {
+        ARTIFACT_PATH = "/usr/share/opensearch/.m2/repository/org/opensearch/client/opensearch-java/${VERSION}"
+        VERSION = "${params.VERSION}"
+    }
+    stages {
+        stage('parameters') {
+            steps {
+                script {
+                    properties([
+                            parameters([
+                                    string(
+                                            defaultValue: '',
+                                            name: 'REF',
+                                            trim: true
+                                    ),
+                                    string(
+                                            name: 'VERSION',
+                                            trim: true
+                                    )
+                            ])
+                    ])
+                }
+            }
+        }
+        stage('Publish to maven local') {
+            steps {
+                // checkout the commit
+                git url: 'https://github.com/opensearch-project/opensearch-java.git', branch: 'main'
+                sh('git checkout ${REF}')
+
+                //publish to maven local
+                sh('./gradlew publishtoMavenLocal')
```

Comment:
I am 99% sure that this should be a shadow publication target with checksums instead.

Reply:
Added a stage to generate checksum. This can be removed once the checksum generation is handled by gradle 
Issue created: https://github.com/opensearch-project/opensearch-java/issues/52